### PR TITLE
[ContextualSaveBar] Apply design language

### PR DESCRIPTION
**Pull request by kyledurand**
### WHY are these changes introduced?

Fixes https://github.com/Shopify/polaris-ux/issues/344

### WHAT is this pull request doing?

image
CSB-fadeup

## 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

1. Hit the details page in storybook
1. Type something into the product title field
1. Dismiss the save bar

Make sure both fade up and in and fade out and down look alright

### 🎩 checklist

* [x] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [x] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
* [ ] Updated the component's `README.md` with documentation changes
* [ ] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
* [ ] For visual design changes, pinged one of @ HYPD, @ mirualves, @ sarahill, or @ ry5n to update the Polaris UI kit

### CODE REVIEW

File: src/components/Frame/Frame.tsx [hunk 1]
```diff
@@ -177,11 +177,16 @@ class FrameInner extends React.PureComponent<CombinedProps, State> {
         </div>
       ) : null;
 
+    const contextualSaveBarClassName = classNames(
+      styles.ContextualSaveBar,
+      unstableGlobalTheming && styles['ContextualSaveBar-globalTheming'],
```
Comment (dleroux) [inline src/components/Frame/Frame.tsx]: Any reason why you're not simply using `.globalTheming`?

**Comment by dleroux** [comment]
What's interesting is the first thing that stood out to me was that the animation was backward. I would have expected the save bar to slide down from the top and leave to the top. Is that just me?

**Comment by dleroux** [comment]
Code looks good. Another thing that's interesting is the fact that the component by itself doesn't show the theming. Is that a problem with the way it's implemented in storybook?

**Comment by kyledurand** [comment]
> Is that a problem with the way it's implemented in storybook

Each component has its own app provider. You'd have to set global theming manually in the readme i.e. https://github.com/Shopify/polaris-react/blob/master/src/components/ContextualSaveBar/README.md#default-contextual-save-bar

**Comment by kyledurand** [comment]
> What's interesting is the first thing that stood out to me was that the animation was backward

The thought behind this is that the changes will be created by a form or change from below the topbar. Those changes push up from where they're being made. I think years of working in and using the web we're used to things sliding on and off screen but with this new design / motion language we're going to be changing that to in and out contextually, if that makes any sense 😄

File: src/components/Frame/Frame.tsx (hunk 1 above)
Comment (kyledurand) [reply src/components/Frame/Frame.tsx]: Something we're going to be removing at some point so I didn't put much thought behind it 🤷‍♂

**Comment by dleroux** [comment]
code cov would be easily achieved by testing this https://codecov.io/gh/Shopify/polaris-react/compare/5f85fbcb02cef7e349ebd9acab8fd9df98968b07...ad59e372cbf6369491e878c687eab4472c7686fa/diff#D7-107. Although not sure it's really necessary since this will be removed eventually.

**Comment by kyledurand** [comment]
image

High contrast mode has been fixed up. Alex has a PR up for adding the border around buttons here: https://github.com/Shopify/polaris-react/pull/2712

@sarahill

File: src/components/Frame/components/CSSAnimation/CSSAnimation.scss [hunk 2]
```diff
@@ -11,3 +11,20 @@
   opacity: 1;
   pointer-events: auto;
 }
+
+.startFadeUp {
+  opacity: 0;
+  will-change: opacity, transform;
+  transform: translateY(#{rem(25px)});
```
Comment (danrosenthal) [inline src/components/Frame/components/CSSAnimation/CSSAnimation.scss]: ✨

File: src/components/Frame/components/ContextualSaveBar/ContextualSaveBar.scss [hunk 3]
```diff
@@ -1,5 +1,5 @@
 @import '../../../../styles/common';
-
+$height: rem(64px);
```
Comment (danrosenthal) [inline src/components/Frame/components/ContextualSaveBar/ContextualSaveBar.scss]: ✨

**Pull request merged**